crates/mbe/src/expander/matcher.rs

```diff
@@ -729,17 +730,16 @@ fn match_meta_var(kind: &str, input: &mut TtIter) -> ExpandResult<Option<Fragmen
     input.expect_fragment(fragment).map(|it| it.map(Fragment::Tokens))
 }
 
-fn collect_vars(buf: &mut Vec<SmolStr>, pattern: &MetaTemplate) {
+fn collect_vars<F: FnMut(SmolStr)>(collector_fun: &mut F, pattern: &MetaTemplate) {
```

Comment:
is it better to use this notation ?
`fn collect_vars(collector_fun: &mut impl FnMut(SmolStr), pattern: &MetaTemplate)`

Reply:
either works, we don't really have any style guidelines on this I think. I personally would probably prefer the `impl` notation here